packages/flutter/test/widgets/focus_traversal_test.dart

```diff
@@ -3786,6 +3786,75 @@ void main() {
     expect(childScope.hasFocus, isTrue);
     expect(nodeA.hasFocus, isFalse);
   });
+
+  testWidgets('GIVEN onFocusNodeCreated is not null '
+      'THEN it is called when the FocusTraversalGroup is built', (WidgetTester tester) async {
+    FocusNode? node;
+    await tester.pumpWidget(
+      FocusTraversalGroup(
+        onFocusNodeCreated: (FocusNode createdNode) => node = createdNode,
+        child: const SizedBox.shrink(),
+      ),
+    );
+    await tester.pumpAndSettle();
+    expect(node, isNotNull);
+  });
+
+  testWidgets(
+    'GIVEN a FocusScope with no focusable descendants '
+    'WHEN the user presses TAB to navigate focus '
+    'THEN focus should skip the scope and land on the next focusable widget without requiring multiple TAB presses',
```

Comment:
Protects the use-case mentioned in the issue https://github.com/flutter/flutter/issues/171516